src/backend/executor/execMain.c

```diff
@@ -876,25 +876,6 @@ standard_ExecutorRun(QueryDesc *queryDesc,
     }
 	PG_CATCH();
 	{
-        /* If EXPLAIN ANALYZE, let qExec try to return stats to qDisp. */
-        if (estate->es_sliceTable &&
-            estate->es_sliceTable->instrument_options &&
-            (estate->es_sliceTable->instrument_options & INSTRUMENT_CDB) &&
-            Gp_role == GP_ROLE_EXECUTE)
-        {
-            PG_TRY();
-            {
-                cdbexplain_sendExecStats(queryDesc);
-            }
-            PG_CATCH();
-            {
-                /* Close down interconnect etc. */
-				mppExecutorCleanup(queryDesc);
-		        PG_RE_THROW();
```

Comment:
I propose to check whether it is possible to implement another solution: remove these three lines from `PG_CATCH();` block, so that the "real" exception (from the upper exception context) is thrown.

As @darthunix mentioned, [cdbexplain_sendExecStats()](https://github.com/arenadata/gpdb/pull/78/files#diff-9ccbf9b96d9eb30e8e709c6bbe57c660L887) currently may use invalid data to produce statistics (as a result, the statistics become corrupted). Thus, if this block with an attempt to send statistics is preserved, extra assertions must be added to `if()`.

Reply:
@leskin-in but still there is a question - what is the reason to return data if an error occurred on one of the segments? In a regular query we’ll return an error but in explain analyze - wrong instrumentation. Usually an error happens on a skewed segment and we hide a real timing in such a masked explain analyze, where we send not finished stats